test/libjulia/Simplifier.cpp

```diff
@@ -139,4 +139,14 @@ BOOST_AUTO_TEST_CASE(mod_and)
 	);
 }
 
+BOOST_AUTO_TEST_CASE(not_applied)
+{
+	CHECK(
+		// The first argument of div is not constant.
+		// keccak256 is not movable.
```

Comment:
This actually highlights a shortcoming of the term "movable". This rule deletes a part of the initial expression, so it can only be applied to side-effect-free expressions. Movable expressions are side-effect free, but they are also context-independent. keccak256 is side-effect free (we usually ignore the side effects on `msize`) but depends on the context and thus is not movable, although the simplification might make sense here. Created an issue: https://github.com/ethereum/solidity/issues/4756